### PR TITLE
Add enum for FatalError + void-alize all functions taking no arguments

### PR DESCRIPTION
**Pull request by madebr**
- Add an enum for the first argument of `FatalError`. The suffix `_S` means it takes one string arguments.
By doing this, I found a few places where an error expects a string argument, but did not pass one.
- Add explicit `void` to all functions accepting no arguments because in C, `void no_args()` declares a function that takes an unspecified (but not variable) number of parameters (and returns nothing). So all calls are valid (according to the prototype) in C.

    I did the `()` -> `(void)` using the following command.
    ```
    find . -iname "*.h" -exec sed -i -E s/\\\(\\\)/\(void\)/g {} \;
    ```
    Afterwards I manually checked the changes and reverted a few unneeded conversions.

### CODE REVIEW

File: src/DETHRACE/common/drmem.c [hunk 1]
```diff
@@ -346,7 +346,7 @@ void CreateStainlessClasses() {
     for (i = 129; i < 246; i++) {
         gStainless_classes[i - 129].res_class = i;
         if (!BrResClassAdd(&gStainless_classes[i - 129])) {
-            FatalError(94);
+            FatalError(kFatalError_OOMCarmageddon_S, gStainless_classes[i - 129].identifier);
```
Comment (madebr) [inline src/DETHRACE/common/drmem.c:349]: Error type 94 needs a string argument.

File: src/DETHRACE/common/depth.c [hunk 2]
```diff
@@ -324,8 +324,8 @@ void InitDepthEffects() {
     PathCat(the_path, gApplication_path, "MATERIAL");
     PathCat(the_path, the_path, "HORIZON.MAT");
     gHorizon_material = BrMaterialLoad(the_path);
-    if (!gHorizon_material) {
-        FatalError(89);
+    if (gHorizon_material == NULL) {
+        FatalError(kFatalError_FindSkyMaterial_S, "HORIZON.MAT"); // 2nd argument added
```
Comment (madebr) [inline src/DETHRACE/common/depth.c:328]: Error type 89 needs a string argument.

File: src/DETHRACE/common/loading.c [hunk 3]
```diff
@@ -1532,7 +1532,7 @@ void GetDamageProgram(FILE* pF, tCar_spec* pCar_spec, tImpact_location pImpact_l
                 }
             }
             if (the_clause->effects[j].type < 0) {
-                FatalError(91);
+                FatalError(kFatalError_UnknownDamageType_S, str);
```
Comment (madebr) [inline src/DETHRACE/common/loading.c:1535]: Error type 91 needs a string argument as well

File: src/DETHRACE/common/world.c [hunk 4]
```diff
@@ -526,14 +526,14 @@ int LoadNModels(tBrender_storage* pStorage_space, FILE* pF, int pCount) {
         PathCat(the_path, gApplication_path, "MODELS");
         PathCat(the_path, the_path, str);
         total = BrModelLoadMany(the_path, temp_array, 2000);
-        if (!total) {
-            FatalError(82);
+        if (total == 0) {
+            FatalError(kFatalError_LoadModelFile_S, str);
```
Comment (madebr) [inline src/DETHRACE/common/world.c:530]: Error type 82 needs a string argument as well.

File: src/DETHRACE/common/world.c [hunk 5]
```diff
@@ -314,14 +314,14 @@ int LoadNPixelmaps(tBrender_storage* pStorage_space, FILE* pF, int pCount) {
             PathCat(the_path, the_path, str);
             total = DRPixelmapLoadMany(the_path, temp_array, COUNT_OF(temp_array));
             if (total == 0) {
-                FatalError(79);
+                FatalError(kFatalError_LoadPixelmapFile_S, str);
```
Comment (madebr) [inline src/DETHRACE/common/world.c:317]: Error type 79 needs a string as well.

File: src/DETHRACE/common/world.c [hunk 6]
```diff
@@ -411,14 +411,14 @@ int LoadNShadeTables(tBrender_storage* pStorage_space, FILE* pF, int pCount) {
         PathCat(the_path, gApplication_path, "SHADETAB");
         PathCat(the_path, the_path, str);
         total = DRPixelmapLoadMany(the_path, temp_array, 50);
-        if (!total) {
-            FatalError(80);
+        if (total == 0) {
+            FatalError(kFatalError_LoadShadeTableFile_S, str);
```
Comment (madebr) [inline src/DETHRACE/common/world.c:415]: Error type 80 needs a string as well.

File: src/DETHRACE/common/world.c [hunk 7]
```diff
@@ -482,14 +482,14 @@ int LoadNMaterials(tBrender_storage* pStorage_space, FILE* pF, int pCount) {
         PathCat(the_path, gApplication_path, "MATERIAL");
         PathCat(the_path, the_path, str);
         total = BrMaterialLoadMany(the_path, temp_array, 200);
-        if (!total) {
-            FatalError(81);
+        if (total == 0) {
+            FatalError(kFatalError_LoadMaterialFile_S, str);
```
Comment (madebr) [inline src/DETHRACE/common/world.c:486]: Error type 81 needs a string as well.

**Review by dethrace-labs — APPROVED** [review]

**Comment by dethrace-labs** [comment]
Nice one! Great that it uncovered some places where we weren't passing the expected string arg too!

**Pull request merged**